cfme/tests/infrastructure/test_provider_discovery.py

```diff
@@ -103,14 +105,18 @@ def test_discover_infra(providers_for_discover, start_ip, max_range):
 
     discover(rhevm, virtualcenter, scvmm, False, start_ip, max_range)
 
-    @pytest.wait_for(num_sec=count_timeout(start_ip, max_range), delay=5)
-    def _wait_for_all_providers():
-        for provider in providers_for_discover:
-            if provider.key not in pytest.store.current_appliance.managed_providers:
+    try:
+        @wait_for_decorator(num_sec=count_timeout(start_ip, max_range), delay=5)
+        def _wait_for_all_providers():
+            for provider in providers_for_discover:
+                if provider.key not in pytest.store.current_appliance.managed_providers:
+                    return False
+            if len(pytest.store.current_appliance.managed_providers) != len(providers_for_discover):
```

Comment:
Though we should never have more than X providers, I think we need to either change the fixture above to ensure there are NO providers, or drop the check for the number of providers. Why did we add this? Is checking for one list against the other not enough?